modin/core/execution/ray/common/deferred_execution.py

```diff
@@ -391,22 +408,24 @@ def _deconstruct_list(
         """
         for obj in lst:
             if isinstance(obj, DeferredExecution):
-                if out_pos := getattr(obj, "out_pos", None):
+                if obj.has_result:
+                    obj = obj.data
```

Comment:
```suggestion
                    out_append(obj.data)
```

I think it would be better to append obj.data in this if branch and remove the continue statements in all the else statements.

Reply:
If `obj.data` is a list, we need to deconstruct it either. Thus, we assign it to `obj` and go to the `if isinstance(obj, ListOrTuple)` check.